File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main()
{
    long card = get_long("Insert your credit card number: ");
    int sum = 0;
    int msum = 0;
    int counter=0;
    int firstdig;
    int firsttwo;
    while (card > 0)
    {
        int lastdig = card%10;
        if (counter%2 == 0)
        {
            msum += lastdig;
        } else
        {
            int doubled = lastdig * 2;
            sum += (doubled % 10) + (doubled / 10);
        }
        card = card/10;
        if (card >= 10 && card < 100)
        {
            firsttwo = card;
        } else if (card >= 1 && card < 10)
        {
            firstdig = card;
        }
        counter++;
    }
    int length = counter;
    int total = sum + msum;
    if (total % 10 == 0)
    {
        if (firstdig == 4 && (length == 13 || length == 16))
        {
            printf("VISA\n");
        } else if ((firsttwo == 34 || firsttwo == 37) && length == 15)
        {
            printf("AMEX\n");
        } else if ((firsttwo >= 51 && firsttwo <= 55) && length == 16)
        {
            printf("MASTERCARD\n");
        } else
        {
            printf("INVALID\n");
        }
    } else
    {
        printf("INVALID\n");
    }
}

File: sentimental-credit/credit.py
import cs50

card = cs50.get_int("Insert your credit card number: ")
sum1 = 0
msum = 0
counter = 0
scard=str(card)
firstdig = int(scard[0])
firsttwo = int(scard[:2])

while card > 0:
    lastdig = card%10
    if counter%2 == 0:
        msum += lastdig
    else:
        doubled = lastdig * 2
        sum1 += (doubled % 10) + (doubled // 10)
    card = card//10
    counter +=1

total = sum1 + msum
if total % 10 == 0:
    if firstdig == 4 and len(scard) in [13, 16]:
        print("VISA")
    elif firsttwo in [34, 37] and len(scard) == 15:
        print("AMEX")
    elif 51 <= firsttwo <= 55 and len(scard) == 16:
        print("MASTERCARD")
    else:
        print("INVALID")
else:
    print("INVALID")
